### DISCUSSION TITLE
Packet Capture with DPDK

### DISCUSSION THREAD

**Discussion by FaisalJAbushanab** [srsran/srsRAN_Project#641] [General Help]
Hi all,

There are multiple options to enable PCAP in srsRAN code at different layers. Is there an option to monitor the OFH when using DPDK? I can't use Wireshark to trace the interface because of DPDK.

I did some basic search and found about this tool:
https://doc.dpdk.org/guides/tools/dumpcap.html

But they say

> The dpdk-dumpcap tool can only be used in conjunction with a primary application which has the packet capture framework initialized already. In DPDK, only the dpdk-testpmd is modified to initialize packet capture framework, other applications remain untouched. So, if the dpdk-dumpcap tool has to be used with any application other than the dpdk-testpmd, user needs to explicitly modify that application to call packet capture framework initialization code.
> 

So does srsRAN support this?

if not, could you please suggest a way to troubleshoot connection issues when trying to connect an O-RU with DPDK on?

**Comment by andrepuschmann** [comment]
We don't support dumpcap.

But there are multiple ways to approach the issue. Firstly we usually use the socket interface when we integrate a new RU or troubleshoot OFH issues. Once the params and configs have been found and potential IOT issues have been solved we switch to DPDK for performance testing and later operation.

Another approach is to configure a mirror port on your switch and do the PCAP on another machine.
